hir/src/formatter/mod.rs

```diff
@@ -0,0 +1,167 @@
+mod document;
+mod print;
+
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
+use core::fmt;
+
+pub use self::document::{concat, const_text, display, flatten, indent, nl, text, Document};
+
+pub trait PrettyPrint {
```

Comment:
Neat! When I was adding `Component` syntax, I thought that it`d be great to try strict Wadler's pretty printer here. And then, I saw you used it in the assembler PR and figured that it will be here as well. Great!